crates/rethnet_evm/src/state/fork.rs

```diff
@@ -79,6 +76,14 @@ impl ForkState {
         self.remote_state.lock().set_block_number(block_number);
     }
 
+    /// Sets the current state root, preventing it from being regenerated.
+    pub fn set_state_root(&mut self, state_root: &B256) {
+        let local_state_root = self.local_state.state_root().unwrap();
```

Comment:
Is this guaranteed to exist? Would prefer an `expect` if yes

Reply:
Resolved by https://github.com/NomicFoundation/hardhat/pull/4418/commits/162b014949e5f86b06eb6bd6111028ee9d934de5